src/schedule.rs

```diff
@@ -166,28 +185,57 @@ impl Schedule {
                     }
                     let hour_range = (Included(Hours::inclusive_min()), Included(hour_start));
 
-                    for hour in self.fields.hours.ordinals().range(hour_range).rev().cloned() {
+                    for hour in self
+                        .fields
+                        .hours
+                        .ordinals()
+                        .range(hour_range)
+                        .rev()
+                        .cloned()
+                    {
                         let minute_start = query.minute_upper_bound();
                         if !self.fields.minutes.ordinals().contains(&minute_start) {
                             query.reset_minute();
                         }
                         let minute_range =
                             (Included(Minutes::inclusive_min()), Included(minute_start));
 
-                        for minute in self.fields.minutes.ordinals().range(minute_range).rev().cloned() {
+                        for minute in self
+                            .fields
+                            .minutes
+                            .ordinals()
+                            .range(minute_range)
+                            .rev()
+                            .cloned()
+                        {
                             let second_start = query.second_upper_bound();
                             if !self.fields.seconds.ordinals().contains(&second_start) {
                                 query.reset_second();
                             }
                             let second_range =
                                 (Included(Seconds::inclusive_min()), Included(second_start));
 
-                            for second in self.fields.seconds.ordinals().range(second_range).rev().cloned()
+                            for second in self
+                                .fields
+                                .seconds
+                                .ordinals()
+                                .range(second_range)
+                                .rev()
+                                .cloned()
                             {
                                 let timezone = before.timezone();
                                 let candidate = if let Some(candidate) = timezone
-                                    .ymd(year as i32, month, day_of_month)
-                                    .and_hms_opt(hour, minute, second)
+                                    // .ymd(year as i32, month, day_of_month)
+                                    // .and_hms_opt(hour, minute, second)
+                                    .with_ymd_and_hms(
```

Comment:
looks like this might interfere with #115

Reply:
@zslayton #112 #115 #116 all look like quality bug fixes. 

I think it would be best if those were merged first then I will rebase this PR.

Reply:
Two of these have been merged, #115 (the most substantive change) is still pending. #123 addressed the current `clippy` and `rustfmt` issues, but did not add those steps to the CI. If you're interested in adding the CI steps as a fresh PR, that might be the quickest path forward.

Reply:
Oops, sorry I missed that this PR included the same fixes for `fmt` and `clippy` as I made in #123, sorry about that! But I didn't add any CI checks (which I should've) so this PR and changes still looks like a good one to merge!

Reply:
I'll try to rebase this today!

Reply:
@Alextopher https://github.com/zslayton/cron/pull/129 just got merged. In case you are currently working on the rebase, I hope this message reaches you before another rebase causes you extra work.